pandas/core/dtypes/common.py

```diff
@@ -1796,6 +1796,11 @@ def pandas_dtype(dtype) -> DtypeObj:
     # try a numpy dtype
     # raise a consistent TypeError if failed
     try:
+        # int is mapped to different types (int32, in64) on Windows and Linux
```

Comment:
hmm, unfortunately, so is numpy array construction, 

existing behaviour
```
>>> arr = np.array([1, 2, 3])  # or arr = np.array([1, 2, 3], dtype=int)
>>> arr.dtype
dtype('int32')
>>>
>>> ser = pd.Series(arr)
>>> ser
0    1
1    2
2    3
dtype: int32
>>>
>>> df = pd.DataFrame(ser)
>>>
>>> df.select_dtypes(include="int")
   0
0  1
1  2
2  3
>>>
```

this workflow would break on Windows with this change?

Reply:
Yes, it wouldn't work on windows:
```
>>>> df.select_dtypes(include="int")

Empty DataFrame
Columns: []
Index: [0, 1, 2]
```

Reply:
In this case, pandas will interpret int as np.int64 and numpy as np.int32 on Windows. Their behavior on Linux will be identical. Though, based on the example in #36596, I see that pandas was mapping integers to np.int64 on Windows by default.

Reply:
we don't want to change this routine at all which has far reaching effects. not averse to a tactical change in .select_dtypes itself.

Reply:
In that case, I can roll back the changes here and add a warning that `int` is ambiguous and `np.int64` or `np.int32` should be used in `df.select_dtypes` instead.  Another option will be to see why pandas does not follow numpy's approach and treats `int` as `int64` on all platforms.